mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -37,6 +37,42 @@ import org.apache.spark.sql.types._
  */
 private[feature] trait RFormulaBase extends HasFeaturesCol with HasLabelCol {
 
+  /**
+   * Param for how to order categories of a string FEATURE column used by `StringIndexer`.
+   * The last category after ordering is dropped when encoding strings.
+   * Supported options: 'frequencyDesc', 'frequencyAsc', 'alphabetDesc', 'alphabetAsc'.
+   * The default value is 'frequencyDesc'. When the ordering is set to 'alphabetDesc', `RFormula`
+   * drops the same category as R when encoding strings.
+   *
+   * The options are explained using an example `'b', 'a', 'b', 'a', 'c', 'b'`:
+   * {{{
+   * +-----------------+---------------------------------------+----------------------------------+
```

Comment:
I think javadoc 8 complains about HTML. It looks this works:

```html
   * <table summary="abc">
   * <tr>
   *   <th>Firstname</th>
   *   <th>Lastname</th>
   *   <th>Age</th>
   * </tr>
   * <tr>
   *   <td>Jill</td>
   *   <td>Smith</td>
   *   <td>50</td>
   * </tr>
   * <tr>
   *   <td>Eve</td>
   *   <td>Jackson</td>
   *   <td>94</td>
   *  </tr>
   * </table>
```

Scaladoc

![2017-05-23 4 28 14](https://cloud.githubusercontent.com/assets/6477701/26342961/19a85edc-3fd5-11e7-8454-5ebaa29bbafd.png)



Javadoc

![2017-05-23 4 27 55](https://cloud.githubusercontent.com/assets/6477701/26342974/23b6ef92-3fd5-11e7-993f-a42c19f5da89.png)

Other errors probably are spurious (please refer https://issues.apache.org/jira/browse/SPARK-20840 which I am fighting with right now).